pkg/syncer/storagepool/storageclasswatch.go

```diff
@@ -214,25 +214,30 @@ func (w *StorageClassWatch) needsRefreshStorageClassCache(ctx context.Context, s
 func (w *StorageClassWatch) refreshStorageClassCache(ctx context.Context) error {
 	log := logger.GetLogger(ctx)
 
-	policyIds := make([]string, 0)
-	policyToSCMap := make(map[string]*storagev1.StorageClass)
-	scClient := w.clientset.StorageV1().StorageClasses()
-	scList, err := scClient.List(ctx, metav1.ListOptions{})
+	policyIDs := make([]string, 0)
+	policyToSCMap := make(map[string]map[string]*storagev1.StorageClass)
+	scList, err := w.clientset.StorageV1().StorageClasses().List(ctx, metav1.ListOptions{})
 	if err != nil {
 		log.Errorf("Failed to query Storage Classes. Err: %+v", err)
 		return err
 	}
+
 	for idx, sc := range scList.Items {
+		idx := idx
 		policyID := getStoragePolicyIDFromSC(&sc)
 		if policyID == "" {
 			continue
 		}
-		policyIds = append(policyIds, policyID)
-		policyToSCMap[policyID] = &scList.Items[idx]
+
+		if _, ok := policyToSCMap[policyID]; !ok {
+			policyToSCMap[policyID] = make(map[string]*storagev1.StorageClass)
+			policyIDs = append(policyIDs, policyID)
+		}
+		policyToSCMap[policyID][scList.Items[idx].Name] = &scList.Items[idx]
```

Comment:
Can we remove dependency on index number of use other way to get mapping?

Reply:
@divyenpatel why would it be a problem?

Reply:
you already have access to SC object in the range, then why do we need to use idx. 
we can directly use sc.name and sc object

Reply:
I've updated the code to use the `sc` object directly.